RocketCast/XMLParser.swift

```diff
@@ -13,31 +13,34 @@ class XMLParser: NSObject {
     
     var element = String()
     var podcast:Podcast?
+    var samePodcast = false
     var tmpEpisode:Episode?
     
     
     init(url: String) {
         super.init()
         if let data = try? Data(contentsOf: URL(string: url)!) {
-        
             podcast = Podcast(context: DatabaseController.getContext())
+            podcast?.addedDate = Date()
             parseData(data)
         } else {
             Log.error("There's nothing in the data from url:\(url)")
         }
     }
     
-   func parseData (_ data:Data) {
+    private func parseData (_ data:Data) {
         let parser = Foundation.XMLParser(data: data)
         parser.delegate = self
         guard parser.parse() else {
             Log.error("Oh shit something went wrong. OS parser failed")
             return
         }
-
-       DatabaseController.saveContext()
+        if (!samePodcast) {
```

Comment:
We wont need this code anymore, by setting the mergePolicy in databaseController.